websockets/websocket.go

```diff
@@ -0,0 +1,133 @@
+package websockets
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/craftamap/hobbit-tracker/hub"
+	"github.com/craftamap/hobbit-tracker/middleware/authtocontext"
+	"github.com/craftamap/hobbit-tracker/middleware/requestcontext"
+	"github.com/craftamap/hobbit-tracker/models"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// Time allowed to write a message to the peer.
+	writeWait = 10 * time.Second
+
+	// Time allowed to read the next pong message from the peer.
+	pongWait = 10 * time.Second
+
+	// Send pings to peer with this period. Must be less than pongWait.
+	// pingPeriod = (pongWait * 9) / 10
+	pingPeriod = (pongWait * 7) / 10
+
+	// Maximum message size allowed from peer.
+	maxMessageSize = 512
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+}
+
+// WebSocketClient is the server-side client of a websocket connection started by a request
+type WebSocketClient struct {
+	log              *logrus.Logger
+	Conn             *websocket.Conn
+	AuthDetails      authtocontext.AuthDetails
+	User             *models.User
+	ServerSideEvents chan hub.ServerSideEvent
+}
+
+func (c *WebSocketClient) handleWriting() {
+	ticker := time.NewTicker(pingPeriod)
+	defer func() {
+		ticker.Stop()
+		c.Conn.Close()
+	}()
+	for {
+		select {
+		case event := <-c.ServerSideEvents:
+			log.Println(event)
+			c.Conn.WriteJSON(event)
+		case <-ticker.C:
+			log.Println("Ping")
```

Comment:
We should remove those, or at least add some kind of identifier to `WebSocketClient`, which can be used as prefix.